### PR TITLE
Feature/prompt coverage in ci pipeline

### PR DESCRIPTION
**Pull request by merwanehamadi**
### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -72,6 +72,19 @@ jobs:
...
+
+      - name: Commit changes
+        run: |
+          git config --global user.name "GitHub Actions Bot"
+          git config --global user.email "github-actions-bot@users.noreply.github.com"
+          git fetch
+          git checkout "${{ github.head_ref }}" || echo "Changes already committed"
+          git add -A
+          git commit -m "Update cassettes" || echo "No changes to commit"
+          git push || echo "Changes already committed"
```
Comment (Pwuts) [inline .github/workflows/ci.yml:87]: Not a fan of committing stuff from CI

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -109,17 +109,18 @@ For more info on running tests, please refer to ["Running tests"](https://docs.a
...
-*Be sure to include the updated cassette in your PR!*
+You can either generate these cassettes locally or by making a pull request to our CI pipeline.
+
 
 When you run Pytest locally:
 
 - If no prompt change: you will not consume API tokens because there are no new OpenAI calls required.
 - If the prompt changes in a way that the cassettes are not reusable:
-    - If no API key, the test fails. It requires a new cassette. So, add an API key to .env.
+    - If no API key, the test fails. It requires a new cassette. So, add an API key to .env or create a PR to generate the cassettes.
```
Comment (Pwuts) [inline CONTRIBUTING.md]: This is not going to work unless they set their API key as a secret on their fork. But for that effort, they might as well just run it locally.

File: tests/integration/agent_factory.py [hunk 3]
```diff
@@ -83,7 +83,7 @@ def writer_agent(agent_test_config, memory_none: NoMemory, workspace: Workspace)
 
     ai_config = AIConfig(
         ai_name="write_to_file-GPT",
-        ai_role="an AI designed to use the write_to_file command to write 'Hello World' into a file named \"hello_world.txt\" and then use the task_complete command to complete the task.",
+        ai_role="an AI designed to use the write_to_file command to write 'Hello World' into a file named \"hello_world.txt\" and then use the task_complete command to complete the task",
```
Comment (Pwuts) [inline tests/integration/agent_factory.py]: ?

File: tests/vcr/vcr_filter.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    filtered_request_without_dynamic_data = replace_timestamp_in_request(
+        filtered_request
+    )
+    return filtered_request_without_dynamic_data
+
+
+def filter_hostnames(request: Any) -> Any:
+    allowed_hostnames: List[str] = [
+        "api.openai.com",
+        "localhost:50337",
```
Comment (Pwuts) [inline tests/vcr/vcr_filter.py:66]: what's this?

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -72,6 +72,19 @@ jobs:
       - name: Run unittest tests with coverage
         run: |
           pytest --cov=autogpt --cov-report term-missing --cov-branch --cov-report xml --cov-report term
+        env:
+          CI: true
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
```
Comment (Pwuts) [inline .github/workflows/ci.yml]: Needs documentation on how to set API key in fork on GitHub

**Review by Pwuts — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (waynehamadi) [reply CONTRIBUTING.md]: yeah exactly they can run it locally or get the cassettes built in the CI.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (waynehamadi) [reply .github/workflows/ci.yml:87]: it's pretty convenient

File: tests/integration/agent_factory.py (hunk 3 above)
Comment (waynehamadi) [reply tests/integration/agent_factory.py]: that was a test

File: tests/vcr/vcr_filter.py (hunk 4 above)
Comment (waynehamadi) [reply tests/vcr/vcr_filter.py:66]: selenium

File: .github/workflows/ci.yml (hunk 5 above)
Comment (waynehamadi) [reply .github/workflows/ci.yml]: Yeah the plan was never really to have people run on their fork.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Pwuts) [reply CONTRIBUTING.md]: And my point is: that's not going to work in this setup. CI on pull requests runs in the context of the fork.

**Pull request closed without merging**